destination.go

```diff
@@ -86,3 +103,62 @@ func (d *Destination) Teardown(ctx context.Context) error {
 	}
 	return nil
 }
+
+// handleInsert create and execute the cql query to insert a row.
+func (d *Destination) handleInsert(ctx context.Context, record sdk.Record) error {
+	table := d.getTableName(record.Metadata)
+	query, vals := d.parser.BuildInsertQuery(record, table)
+	err := d.session.Query(query, vals...).Exec()
+	if err != nil {
+		return fmt.Errorf("error while inserting data: %w", err)
+	}
+
+	return nil
+}
+
+// handleUpdate create and execute the cql query to update a row.
+func (d *Destination) handleUpdate(ctx context.Context, record sdk.Record) error {
+	table := d.getTableName(record.Metadata)
+	query, vals := d.parser.BuildUpdateQuery(record, table)
+	err := d.session.Query(query, vals...).Exec()
+	if err != nil {
+		return fmt.Errorf("error while updating data: %w", err)
+	}
+
+	return nil
+}
+
+// handleDelete create and execute the cql query to delete a row.
+func (d *Destination) handleDelete(ctx context.Context, record sdk.Record) error {
+	table := d.getTableName(record.Metadata)
+	query, vals := d.parser.BuildDeleteQuery(record, table)
+	err := d.session.Query(query, vals...).Exec()
+	if err != nil {
+		return fmt.Errorf("error while deleting data: %w", err)
+	}
+
+	return nil
+}
+
+// validateStructuredRecord return an error if the record key or payload is not structured.
+func (d *Destination) validateStructuredRecord(record sdk.Record) error {
```

Comment:
Maybe make it possible to accept raw JSON payloads as well? We might see those more often than not because of the timestamp issue https://github.com/ConduitIO/conduit/issues/866.

Reply:
this was marked as future work in the research doc (not that far of a future, more like a second step after finishing structured data), will create an issue for it to be tackled after!

Reply:
JSON payloads can be transformed into structured data with a processor though.

Reply:
JSON payloads will be transferred as RawData right? IIUC, I was planning to check if the rawData is parsable as JSON, if it is, then this will be handled differently than just rawData.
I think it's a good idea to first support raw JSON payloads, especially with the timestamp issue, so I'll work on that separately from pure rawData, and will tackle that after the integration tests!

Reply:
Yes, exactly. Parsing the JSON can already be done in the processor and even though the usage would be easier if the connector would parse the JSON itself I think it's a slippery slope where we start packing functionality in the connectors just to make them easy to use. If something can be done in a processor IMO it should be avoided in connectors.

The timestamp issue is not resolved if we parse it here or in a processor, the timestamp will be parsed as a string in both cases.